specification/batch/Batch/routes.tsp

```diff
@@ -0,0 +1,2755 @@
+      @query
+      $select?: string;
+    },
+    BatchResponseHeaders & NodeVMExtensionList,
+    {},
+    BatchError
+  >;
+}
+
+// headers ////////////////////
+
+#suppress "@azure-tools/typespec-azure-core/casing-style" "The names of Model types must use camelCase"
+@doc("The request to the Batch service was successful.")
+model PoolRemoveNodesResponseHeaders {
+  ...BatchResponseHeaders;
+
+  @doc("A process exit code.")
+  @statusCode
+  code: "202";
+
+  @doc("The OData ID of the resource to which the request applied")
+  @header("DataServiceId")
+  DataServiceId: string;
+}
+
+#suppress "@azure-tools/typespec-azure-core/casing-style" "The names of Model types must use camelCase"
+@doc("The request to the Batch service was successful.")
+model PoolUpdatePropertiesResponseHeaders {
+  ...BatchResponseHeaders;
+
+  @doc("A process exit code.")
+  @statusCode
+  code: "204";
+
+  @doc("The OData ID of the resource to which the request applied")
+  @header("DataServiceId")
+  DataServiceId: string;
+}
+
+alias PoolStopResizeResponseHeaders = PoolResizeResponseHeaders;
+
+#suppress "@azure-tools/typespec-azure-core/casing-style" "The names of Model types must use camelCase"
+@doc("The request to the Batch service was successful.")
+model PoolResizeResponseHeaders {
+  ...BatchResponseHeaders;
+
+  @doc("A process exit code.")
+  @statusCode
+  code: "202";
+
+  @doc("The OData ID of the resource to which the request applied")
+  @header("DataServiceId")
+  DataServiceId: string;
+}
+
+#suppress "@azure-tools/typespec-azure-core/casing-style" "The names of Model types must use camelCase"
+@doc("Common header values for pool requests")
+model FileResponse {
+  @doc("A process exit code.")
+  @statusCode
+  code: "200";
+
+  @header
+  @doc("The file creation time.")
+  @format("date-time-rfc1123")
+  `ocp-creation-time`?: string;
+
+  @header
+  @doc("Whether the object represents a directory.")
+  `ocp-batch-file-isdirectory`: boolean;
+
+  @header
+  @doc("The URL of the file.")
+  `ocp-batch-file-url`: string;
+
+  @header
+  @doc("The file mode attribute in octal format.")
+  `ocp-batch-file-mode`: string;
+
+  @header
+  @doc("The length of the file.")
+  `Content-Length`: int64;
+}
+
+
+/////////////////////////////////////  Headers models ///////////////////////////////////
+
+@doc("Common header values for pool requests")
+model BatchPoolHeaders extends BatchMatchHeaders {
+  @doc("The ID of the Pool to get.")
+  @path
+  poolId: string;
+}
+
+#suppress "@azure-tools/typespec-azure-core/casing-style" "The names of Model types must use camelCase"
+@doc("Common header values for pool requests")
+model BatchMatchHeaders extends BatchModifiedSinceHeaders {
+  @doc("""
+An ETag value associated with the version of the resource known to the client.
+The operation will be performed only if the resource's current ETag on the
+service exactly matches the value specified by the client.
+""")
+  @header
+  `If-Match`?: string;
```

Comment:
Just a reminder: if 304/412 is a valid response code, you should explicitly define it as response. We don't magically add those codes as a valid response when you define etag header. Put it another way, we only take the status codes you defined in your operations as valid response codes.